src/App.tsx

```diff
@@ -26,70 +31,116 @@ type ReorderOptions = {
+      visibleGoods.sort(
+        (good1, good2) => good1.localeCompare(good2),
+      );
+      break;
 
-  return visibleGoods;
-}
+    case SortType.LENGTH:
+      visibleGoods.sort(
+        (good1, good2) => good1.length - good2.length,
+      );
+      break;
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+    default:
+      break;
+  }
 
-      <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
-      </ul>
-    </div>
-  );
+  return isReversed
+    ? visibleGoods.reverse()
+    : visibleGoods;
+}
+
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
 };
+
+export class App extends Component<{}, State> {
+  state = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  handleSort = (sortType: SortType) => {
+    this.setState({ sortType });
+  };
+
+  handleReverse = () => {
+    this.setState((state) => ({
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  handleReset = () => {
+    this.setState({
+      sortType: SortType.NONE,
+      isReversed: false,
+    });
+  };
+
+  render() {
+    const { sortType, isReversed } = this.state;
+    const validGoods = getReorderedGoods(goodsFromServer, this.state);
+    const canReset = (sortType !== SortType.NONE) || isReversed;
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={classNames(
+              'button is-info',
+              { 'is-light': sortType !== SortType.ALPHABET },
+            )}
+            onClick={() => this.handleSort(SortType.ALPHABET)}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={classNames(
+              'button is-success',
+              { 'is-light': sortType !== SortType.LENGTH },
+            )}
+            onClick={() => this.handleSort(SortType.LENGTH)}
+          >
+            Sort by length
+          </button>
```

Comment:
Would be nice to create an array from `SortType` enum (use Object.values) and map it here instead of code repeating